src/audio/host-zephyr.c

```diff
@@ -1027,119 +852,8 @@ void host_zephyr_reset(struct host_data *hd, uint16_t state)
 	hd->sink = NULL;
 }
 
-static int host_reset(struct comp_dev *dev)
-{
-	struct host_data *hd = comp_get_drvdata(dev);
-
-	comp_dbg(dev, "host_reset()");
-	/* remove callback first for host reset */
-	if (hd->chan)
-		notifier_unregister(dev, hd->chan, NOTIFIER_ID_DMA_COPY);
-
-	host_zephyr_reset(hd, dev->state);
-	dev->state = COMP_STATE_READY;
-
-	return 0;
-}
-
-/* copy and process stream data from source to sink buffers */
-int host_zephyr_copy(struct host_data *hd, struct comp_dev *dev)
-{
-	return hd->copy(hd, dev);
-}
-
 /* copy and process stream data from source to sink buffers */
-static int host_copy(struct comp_dev *dev)
+int host_zephyr_copy(struct host_data *hd, struct comp_dev *dev, copy_callback_t cb)
 {
-	struct host_data *hd = comp_get_drvdata(dev);
-
-	if (dev->state != COMP_STATE_ACTIVE)
-		return 0;
-
-	return host_zephyr_copy(hd, dev);
+	return hd->copy(hd, dev, cb);
 }
-
-static int host_get_attribute(struct comp_dev *dev, uint32_t type,
-			      void *value)
-{
-	struct host_data *hd = comp_get_drvdata(dev);
-
-	switch (type) {
-	case COMP_ATTR_COPY_TYPE:
-		*(enum comp_copy_type *)value = hd->copy_type;
-		break;
-	case COMP_ATTR_COPY_DIR:
-		*(uint32_t *)value = hd->ipc_host.direction;
-		break;
-	default:
-		return -EINVAL;
-	}
-
-	return 0;
-}
-
-static int host_set_attribute(struct comp_dev *dev, uint32_t type,
-			      void *value)
-{
-	struct host_data *hd = comp_get_drvdata(dev);
-
-	switch (type) {
-	case COMP_ATTR_COPY_TYPE:
-		hd->copy_type = *(enum comp_copy_type *)value;
-		break;
-	case COMP_ATTR_HOST_BUFFER:
-		hd->host.elem_array = *(struct dma_sg_elem_array *)value;
-		break;
-	default:
-		return -EINVAL;
-	}
-
-	return 0;
-}
-
-static uint64_t host_get_processed_data(struct comp_dev *dev, uint32_t stream_no, bool input)
-{
-	struct host_data *hd = comp_get_drvdata(dev);
-	uint64_t ret = 0;
-	bool source = dev->direction == SOF_IPC_STREAM_PLAYBACK;
-
-	/* Return value only if direction and stream number match.
-	 * The host supports only one stream.
-	 */
-	if (stream_no == 0 && source == input)
-		ret = hd->total_data_processed;
-
-	return ret;
-}
-
-static const struct comp_driver comp_host = {
-	.type	= SOF_COMP_HOST,
-	.uid	= SOF_RT_UUID(host_uuid),
-	.tctx	= &host_tr,
```

Comment:
You are making host-zephyr from a device to a small lib, so at least, title may need change, it is not simply remove code, with your changes, host-zephyr.c may change to host-zephyr-lib.c

Reply:
I don't think we need to change the name of the file here. lib could indicate that it is built as part of a library.